utils.js

```diff
@@ -54,13 +48,11 @@ exports.normalizeScopeSets = function(scopesets) {
  * Also expressed as ('a' and 'b') or 'c'.
  */
 exports.scopeMatch = function(scopePatterns, scopesets) {
-  var scopesets = exports.normalizeScopeSets(scopesets);
+  exports.validateScopeSets(scopesets);
   if (typeof(scopePatterns) == 'string') {
     scopePatterns = [scopePatterns];
   }
```

Comment:
Looks good.  The reason I used `validateScopeSets` was to allow `api.declare` to apply the same logic.  You're right that it adds a few extra iterations, although a good enough JIT could fix that ;)